R/parser.R

```diff
@@ -915,6 +915,173 @@ parse_jira <- function(json_path){
 
   return(parsed_issues_comments)
 }
+#' New Parse Jira issue and comments
+#'
+#' @param json_path path to jira json (issues or issues with comments) obtained from the improved Jira Dowmloader.
+#' @return A named list of two named elements ("issues", and "comments"), each containing a data.table.
+#' Note the comments element will be empty if the downloaded json only contain issues. Also,
+#' the jira json naming convention is as follows: "(ProjectKey)_issues_(UNIXTIMElowerbound)_(UNITITMEupperbound).json"
+#' or "(ProjectKey)_issue_comments_(UNIXTIMElowerbound)_(UNIXTIMEupperbound).json"
+#' @export
+#' @family parsers
+parse_jira_new <- function(json_path){
+
+  file_list <- list.files(json_path)
+
+  if (identical(file_list, character(0))){
+    stop(stringi::stri_c("cannot open the connection"))
+  }
+
+  # Comments list parser. Comments may occur on any json issue.
+  jira_parse_comment <- function(comment){
+    parsed_comment <- list()
+    parsed_comment[["comment_id"]] <- comment[["id"]]
+
+    parsed_comment[["comment_created_datetimetz"]] <- comment[["created"]][[1]]
+    parsed_comment[["comment_updated_datetimetz"]] <- comment[["updated"]][[1]]
+
+    parsed_comment[["comment_author_id"]] <- comment[["author"]][["name"]][[1]]
+    parsed_comment[["comment_author_name"]] <- comment[["author"]][["displayName"]][[1]]
+    parsed_comment[["comment_author_timezone"]] <- comment[["author"]][["timeZone"]][[1]]
+
+    parsed_comment[["comment_author_update_id"]] <- comment[["updateAuthor"]][["name"]][[1]]
+    parsed_comment[["comment_author_update_name"]] <- comment[["updateAuthor"]][["displayName"]][[1]]
+    parsed_comment[["comment_author_update_timezone"]] <- comment[["updateAuthor"]][["timeZone"]][[1]]
+
+    parsed_comment[["comment_body"]] <- comment[["body"]][[1]]
+
+    return(parsed_comment)
+  }
+
+  # Issues parser
+  jira_parse_issues <- function(jira_file){
```

Comment:
This function name is misleading, as it also parse comments. Where is the comparison to other functions API on Bugzilla and GitHub? I'd like to compare to decide on having a function parse_issues_folder and parse_issues vesus what you have right now.